runtime/compiler/env/ClassLoaderTable.cpp

```diff
@@ -183,28 +204,48 @@ TR_PersistentClassLoaderTable::associateClassLoaderWithClass(J9VMThread *vmThrea
       return;
 
 #if defined(J9VM_OPT_JITSERVER)
-   bool useAOTCache = _persistentMemory->getPersistentInfo()->getJITServerUseAOTCache();
-   const J9UTF8 *nameStr = J9ROMCLASS_CLASSNAME(((J9Class *)clazz)->romClass);
-   const uint8_t *name = J9UTF8_DATA(nameStr);
-   uint16_t nameLength = J9UTF8_LENGTH(nameStr);
+   auto romClass = ((J9Class *)clazz)->romClass;
+   auto romName = J9ROMCLASS_CLASSNAME(romClass);
+   const uint8_t *name = J9UTF8_DATA(romName);
+   size_t nameLength = J9UTF8_LENGTH(romName);
 #endif /* defined(J9VM_OPT_JITSERVER) */
 
-   uintptr_t classChainOffset = _sharedCache->rememberClass(clazz);
-   if (TR_SharedCache::INVALID_CLASS_CHAIN_OFFSET == classChainOffset)
+   void *chain = NULL;
+   if (_sharedCache)
       {
+      uintptr_t chainOffset = _sharedCache->rememberClass(clazz);
+      if (TR_SharedCache::INVALID_CLASS_CHAIN_OFFSET == chainOffset)
+         {
 #if defined(J9VM_OPT_JITSERVER)
-      if (useAOTCache && TR::Options::getVerboseOption(TR_VerboseJITServer))
-         TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
-            "ERROR: Failed to get class chain for %.*s loaded by %p",
-            nameLength, (const char *)name, loader
-         );
+         if (useAOTCache && TR::Options::getVerboseOption(TR_VerboseJITServer))
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITServer,
+               "ERROR: Failed to get class chain for %.*s loaded by %p",
+               nameLength, (const char *)name, loader
+            );
 #endif /* defined(J9VM_OPT_JITSERVER) */
+         }
+      else
+         {
+         chain = _sharedCache->pointerFromOffsetInSharedCache(chainOffset);
+         TR_ASSERT(_sharedCache->isPointerInSharedCache(chain), "Class chain must be in SCC");
+         }
+      }
+
+   // If we are using the JITServer AOT cache, we need to remember the name of clazz with or without chain.
+   // If we aren't using it, there is no point in continuing.
+   if (!chain && !useAOTCache)
       return;
```

Comment:
Don't we miss the return here?

Reply:
I think it's there, in line 237 just below the diff?

https://github.com/eclipse-openj9/openj9/pull/18839/files#diff-f8ebc6246869c77a730d272b736cc0a8f2778232b1ce5ebc8eb4e1776db91194R237

Reply:
Based on our discussion I added early returns to the `if (_sharedCache)` block so we don't bother continuing if an SCC isn't present or we can't find the chain offset. These will be removed when in the main JITServer AOT cache PR.